modules/services/repositories/src/main/java/au/com/shiftyjelly/pocketcasts/repositories/user/UserManager.kt

```diff
@@ -108,6 +108,7 @@ class UserManagerImpl @Inject constructor(
 
         settings.setMarketingOptIn(false)
         settings.setMarketingOptInNeedsSync(false)
+        settings.setEndOfYearModalHasBeenShown(false)
```

Comment:
This reset is still needed to matches iOS behavior to show the modal again when a user login (or creates an account): https://github.com/Automattic/pocket-casts-ios/pull/474/commits/2903fada733fb0b86a9543726aee27f269b28cb2

The difference now is that it is done on explicit signout rather than when `SignInState` is `SignedOut` when the app is restarted.

Reply:
I ran across 2 edge cases here that I wasn't sure about.

#### 1. Signing in frequently doesn't trigger the dialog

Just noting that this only works if the user does not close the app between signing out and log-in/creating-an-account. If they do restart the app, then the pop-up displays when the app restarts with them signed out, but does not display when they sign in.

Similarly, if a user:
1. Freshly installs the app
2. Listens to 5+ minutes of podcast without signing in
3. Restarts the app
4. They will see the stats popup
5. Signs in with an account that adds a bunch of new listening history
6. Restarts the app
7. They will not see the pop-up

These seem a bit like edge cases, so I don't think they're blockers at all. And maybe the current behavior is actually what we want, I just wasn't sure since a user who signs in could have a lot of new data.

#### 2. Signing in and then Signing out immediately shows the dialog

1. Open the app with more than 5 minutes of listening time
2. See the End of Year dialog
3. Sign into an account
4. Sign out of the account
5. Observe that the EOY dialog is immediately shown after sign out
6. Sign into an account
7. Observe that the EOY dialog is not shown

Would we rather the dialog show at step 7 instead of step 5?